pkg/golinters/exhaustruct.go

```diff
@@ -0,0 +1,32 @@
+package golinters
+
+import (
+	"strings"
+
+	"github.com/GaijinEntertainment/go-exhaustruct/pkg/analyzer"
+	"golang.org/x/tools/go/analysis"
+
+	"github.com/golangci/golangci-lint/pkg/config"
+	"github.com/golangci/golangci-lint/pkg/golinters/goanalysis"
+)
+
+func NewExhaustruct(settings *config.ExhaustructSettings) *goanalysis.Linter {
+	a := analyzer.Analyzer
+
+	var cfg map[string]map[string]interface{}
+	if settings != nil {
+		cfg = map[string]map[string]interface{}{
+			a.Name: {
+				"include": strings.Join(settings.Include, ","),
+				"exclude": strings.Join(settings.Exclude, ","),
```

Comment:
I think there are problems here: `,` is something that can be used in a regular expression (ex: `\d{1,3}`).
So the parsing by the linter will produce unexpected behavior in this case.

<details><summary>parsing problem example</summary>

```yml
linters:
  disable-all: true
  enable:
    - exhaustruct

linters-settings:
  exhaustruct:
    include: ".*\\.ExhaustructCustom2{1,}$"
```

```console
$ ./golangci-lint run
$ 
```

```go

type ExhaustructCustom22 struct {
	A string
	B int
	c bool
	D float64
	E time.Time
}

func exhaustructCustom2() {
	_ = ExhaustructCustom22{
		A: "a",
		B: 0,
		c: false,
		D: 1.0,
		E: time.Now(),
	}

	_ = ExhaustructCustom22{
		A: "a",
		c: false,
		D: 1.0,
		E: time.Now(),
	}

	_ = ExhaustructCustom22{
		A: "a",
		c: false,
		E: time.Now(),
	}

	_ = ExhaustructCustom22{
		A: "a",
		B: 0,
		D: 1.0,
		E: time.Now(),
	}
}
```


</details>

Reply:
Oof. Haven't thought about quantifiers🤔
Maybe replace delimiter with semicolon, it is not used in regexp syntax, nor in golang (at least Im unable to recall)

Reply:
I think it's better to not play with separators if you can.
You can forward the configuration as slice or struct.

Reply:
Anyways I'll make a change later today

Reply:
Updated linter and adopted to changes here.
Now regexps passed as-is, without concatenation.

Reply:
Can you remove the panic?

Reply:
Actually, I've already made such change, but yet away from wifi, I'll push it in 4-5 hours.
Yet only I haven't figured how to return an error dirung linter initialization (for the case provided strings are not valid regexp) 🤔

Reply:
@ldez advices are appreciated =)

Reply:
Sorry for the very late answer, time is always a problem.

I added a commit to handle the error, can you create a new tag for your linter?

Reply:
Sure.

Reply:
v2.1.0 is there

Reply:
thank you!